File: antigravity/a-vanilla/run-3/src/orm/Loan.php
<?php

use Tina4\ORM;

class Loan extends ORM
{
    public string $tableName = "loans";
    public string $primaryKey = "id";
    public bool $autoMap = true;

    public int $id;
    public int $bookId;
    public int $memberId;
    public string $borrowDate;
    public string $dueDate;
    public ?string $returnedDate = null;
    public string $createdAt;
    public string $updatedAt;

    /**
     * Get the book borrowed in this loan
     * @return Book|null
     */
    public function book(): ?Book
    {
        return $this->belongsTo(Book::class, "book_id");
    }

    /**
     * Get the member who made this loan
     * @return Member|null
     */
    public function member(): ?Member
    {
        return $this->belongsTo(Member::class, "member_id");
    }
}

File: antigravity/a-vanilla/run-3/src/orm/Loan.py
from tina4_python.orm import ORM, IntegerField, StringField, ForeignKeyField
from src.orm.Book import Book
from src.orm.Member import Member
from src.orm.User import User

class Loan(ORM):
    table_name = "loans"
    id = IntegerField(primary_key=True, auto_increment=True)
    book_id = ForeignKeyField(to=Book, related_name="loans")
    member_id = ForeignKeyField(to=Member, related_name="loans")
    borrow_date = StringField()
    due_date = StringField()
    returned = IntegerField()
    returned_date = StringField()
    staff_id = ForeignKeyField(to=User, related_name="loans")
